src/systemctl/systemctl-start-special.c

```diff
@@ -201,14 +201,17 @@ int verb_start_special(int argc, char *argv[], void *userdata) {
                 case ACTION_KEXEC:
                 case ACTION_HALT:
                 case ACTION_SOFT_REBOOT:
-                        if (arg_when == 0)
+                        if (arg_when == 0) {
                                 r = logind_reboot(a);
-                        else
+                                if (r >= 0 || IN_SET(r, -EACCES, -EOPNOTSUPP, -EINPROGRESS))
+                                        /* The latter indicates that the requested operation requires auth,
+                                         * is not supported or already in progress, in which cases we ignore the error. */
+                                        return r;
+                        } else {
                                 r = logind_schedule_shutdown(a);
-                        if (r >= 0 || IN_SET(r, -EACCES, -EOPNOTSUPP, -EINPROGRESS))
```

Comment:
You might want to see https://github.com/systemd/systemd/issues/17575 and https://github.com/systemd/systemd/pull/18010

Reply:
Well, this changes only `systemctl --when=`, i.e. our new-ish stuff rather than the sysv-compat `shutdown`. _Maybe_ the behavior was sensible for the latter, whose time spec is restricted within 24 hours, but not here. Therefore I really think the new logic is the way to go.

I mean, I introduced `--when=` originally, and I nowadays strongly feel like that it was a mistake to take this fallback logic from sysv compat code. I'd also add support for `sleep --when=` in the future, and it seems like a good time to first get rid of such problematic behavior from systemctl.

cc @poettering